lib/AST/ASTPrinter.cpp

```diff
@@ -285,14 +285,17 @@ void ASTPrinter::printEscapedStringLiteral(StringRef str) {
   printTextImpl(escapeBuf.str());
 }
 
-void ASTPrinter::printTypeRef(Type T, const TypeDecl *RefTo, Identifier Name) {
+void ASTPrinter::printTypeRef(Type T, const TypeDecl *RefTo, Identifier Name,
+                              bool printedModuleContext) {
   PrintNameContext Context = PrintNameContext::Normal;
```

Comment:
Maybe you should pass in the initial value of `Context` and have it default to `PrintNameContext::Normal`, but have `TypePrinter::printQualifiedType()` pass `PrintNameContext::ModuleQualifiedType` instead when appropriate.

Reply:
I think passing the context makes sense here.